src/cvsim/fit_curve.py

```diff
@@ -91,10 +115,37 @@ def _ensure_positive_or_none(param: str, value: float | None):
         if value is not None and value <= 0.0:
             raise ValueError(f"'{param}' must be > 0.0 or None")
 
+    @staticmethod
+    def _ensure_open_unit_interval_or_none(param: str, value: float | None):
+        if value is not None and not 0.0 < value < 1.0:
+            raise ValueError(f"'{param}' must be between 0.0 and 1.0, or None")
+
     @staticmethod
     def _non_none_dict(mapping: dict):
         return {k: v for k, v in mapping.items() if v is not None}
 
+    @staticmethod
+    def _fit_var_checker(fit_vars: dict, fit_default_vars: dict) -> dict:  # TODO rename
+        # take fit_vars dict, for each in it, replace the initial guess/bounds if specified
+        for param, value in fit_vars.items():
+            if isinstance(value, float | int):
+                # Initial guess
+                fit_default_vars[param][0] = value
+            elif isinstance(value, tuple) and len(value) == 2:
+                # Lower and upper bound
+                if value[0] >= value[1]:
+                    raise ValueError("Lower bound must be lower than upper bound")
+                fit_default_vars[param][1] = value[0]
+                fit_default_vars[param][2] = value[1]
+            elif isinstance(value, tuple) and len(value) == 3:
+                if value[1] >= value[2]:
```

Comment:
We should check that the estimate is in the range as well:
```suggestion
                if not value[1] < value[0] < value[2]:
```